### PR TITLE
Added code for rmq problem with updates.

### PR DESCRIPTION
**Pull request by FR4NKESTI3N**
**Fixes issue:**
Moved ada segment tree code with the rest of segment tree code.

**Changes:**
Added code for rmq problem with updates.

### CODE REVIEW

**Comment by FR4NKESTI3N** [comment]
Alright, I'll make a class SegmentTree. Should make next commit in soon. Should I add spaces around '->'?

**Comment by arnavb** [comment]
Not for templates, no, but for `<<` and `>>` yes.

**Comment by FR4NKESTI3N** [comment]
I have combined the methods under a SegmentTree class and added space around the `<<` where I missed them.

File: code/data_structures/src/tree/space_partitioning_tree/segment_tree/segment_tree_rmq_with_update.cpp [hunk 1]
```diff
@@ -20,114 +20,149 @@ struct Node
...
     int n = 8;
     std::vector<int> array = {5, 4, 3, 2, 1, 0, 7, 0};
 
-    Node *root = new Node;
-    build(array, root, 0, n - 1);
+    SegmentTree st(array);
+
+    std::cout << "Array:\n";
+    for (int i = 0; i < n; ++i)
+        std::cout<<st.array[i] << ' ';
```
Comment (arnavb) [inline code/data_structures/src/tree/space_partitioning_tree/segment_tree/segment_tree_rmq_with_update.cpp]: Add spaces around the operators.

**Comment by FR4NKESTI3N** [comment]
I added them in the commit I made right after my comment.

**Comment by arnavb** [comment]
djldk 2

**Pull request merged**